source/backend/api/Services/LeaseService.cs

```diff
@@ -471,5 +489,67 @@ private void AppendNewItemsToChecklist(PimsLease lease, ref List<PimsLeaseCheckl
                 }
             }
         }
+
+        private void ValidateRenewalDates(PimsLease lease, ICollection<PimsLeaseRenewal> renewals)
+        {
+            if (lease.LeaseStatusTypeCode != PimsLeaseStatusTypes.ACTIVE)
+            {
+                return;
+            }
+
+            List<Tuple<DateTime, DateTime>> renewalDates = new();
+
+            foreach (var renewal in renewals)
+            {
+                if (renewal.IsExercised == true)
+                {
+                    if (renewal.CommencementDt.HasValue && renewal.ExpiryDt.HasValue)
+                    {
+                        renewalDates.Add(new Tuple<DateTime, DateTime>(renewal.CommencementDt.Value, renewal.ExpiryDt.Value));
+                    }
+                    else
+                    {
+                        throw new BusinessRuleViolationException("Excercised renewals must have a commencement date and expiry date");
+                    }
+                }
+            }
+
+            // Sort agreement dates/renewals by start date.
+            renewalDates.Sort((a, b) => DateTime.Compare(a.Item1, b.Item1));
+
+            DateTime currentEndDate;
+
+            if (lease.OrigStartDate.HasValue && lease.OrigExpiryDate.HasValue)
+            {
+                var agreementStart = lease.OrigStartDate.Value;
+                var agreementEnd = lease.OrigExpiryDate.Value;
+                currentEndDate = agreementEnd;
+                if (DateTime.Compare(agreementEnd, agreementStart) <= 0)
+                {
+                    throw new BusinessRuleViolationException("The lease commencement date must be before its expiry date");
+                }
+            }
+            else
+            {
+                throw new BusinessRuleViolationException("Active leases must have commencement and expiry dates");
```

Comment:
is this true? I think they just need commencement dates.

Reply:
Yeah, confirmed by julian. Check above reply